### PR TITLE
Fix metadata template object for transaction

### PR DESCRIPTION
**Pull request by josenavas**
This PR fixes the files under `qiita_db/metadata_template`

Few comments:
 - The functions `to_file` and `create_qiime_mapping_file` were missing the `encoding` parameter for `to_csv`
 - The `SampleTemplate.delete` function was checking for `RawData` to allow deletion of the Sample Template. This is incorrect as it should be checking for `PrepTemplate`.

### CODE REVIEW

File: qiita_db/metadata_template/base_metadata_template.py [hunk 1]
```diff
@@ -208,16 +204,16 @@ def _to_dict(self):
         dict of {str: str}
             A dictionary of the form {category: value}
         """
-        conn_handler = SQLConnectionHandler()
-        d = dict(conn_handler.execute_fetchone(
-            "SELECT * from qiita.{0} WHERE "
-            "sample_id=%s".format(self._dynamic_table),
-            (self._id, )))
+        with TRN:
+            sql = "SELECT * from qiita.{0} WHERE sample_id=%s".format(
```
Comment (ElDeveloper) [inline qiita_db/metadata_template/base_metadata_template.py]: from -> FROM

File: qiita_db/metadata_template/prep_template.py [hunk 2]
```diff
@@ -212,62 +204,65 @@ def delete(cls, id_):
...
+        with TRN:
+            table_name = cls._table_name(id_)
+
+            if not cls.exists(id_):
+                raise QiitaDBUnknownIDError(id_, cls.__name__)
+
+            sql = """SELECT EXISTS(
+                        SELECT * FROM qiita.prep_template_preprocessed_data
+                        WHERE prep_template_id=%s)"""
+            args = [id_]
```
Comment (ElDeveloper) [inline qiita_db/metadata_template/prep_template.py]: noice

**Comment by ElDeveloper** [comment]
Just one tiny comment, :+1: regardless.

**Comment by antgonza** [comment]
Yup :+1: once that minor comment is addressed.

File: qiita_db/metadata_template/base_metadata_template.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/metadata_template/base_metadata_template.py]: Fixed

File: qiita_db/metadata_template/prep_template.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/metadata_template/prep_template.py]: :smile:

**Comment by josenavas** [comment]
@ElDeveloper @antgonza comments addressed.

**Comment by ElDeveloper** [comment]
@josenavas there's a real failure in your latest build.

**Comment by josenavas** [comment]
Thanks for the heads up @ElDeveloper I notice that this error is also in the upstream/transaction branch so I will pull and probably there is something that fail when everything got together...

**Comment by josenavas** [comment]
Tests should pass now. Let them execute to make sure :smile:

**Comment by josenavas** [comment]
:boom:

**Pull request merged**